### DISCUSSION TITLE
binary image classification with cnn

### DISCUSSION THREAD

**Discussion by eishwezin20** [tensorflow/agents#886] [Q&A] [answered]
I am trying to classify if a given image is a woodpile or not. So my classes are wood and none_wood. None_wood class contains different photos including documents, numbers, persons etc. My train set consist of 4655 wood and 4566 none_wood images(I used augmentor to increase the size of none_wood class) and test set includes 1139 images of each class.

My model is as following:

classifier = Sequential()
classifier.add(Convolution2D(32,(3,3),input_shape = (175,130,3), activation = 'relu'))
classifier.add(MaxPooling2D(pool_size = (2,2)))
classifier.add(Convolution2D(64,(3,3),input_shape = (175,130,3), activation = 'relu'))
classifier.add(MaxPooling2D(pool_size = (2,2)))
classifier.add(Flatten())
classifier.add(Dense(128, activation = 'relu'))
classifier.add(Dense(1, activation = 'sigmoid'))
classifier.compile(optimizer = 'adam', loss = 'binary_crossentropy', metrics = ['accuracy'])

I tried different epochs up to 10, I added callback to prevent overfit, I did data augmentation with ImageDataGenerator, there is no imbalance in the dataset, I tried different optimizers, I tried different loss functions but there is no change.

What do you suggest? I could not understand what is the exact problem and why the FP and TN values are 0 in confusion matrix?

**Comment by Mattral** [comment, marked as answer]
there is a duplication of the Convolution2D layer with the same parameters in your model architecture. You should have a different set of layers for building a convolutional neural network (CNN) properly. Here's a revised version of your model:
from tensorflow.keras.models import Sequential
from tensorflow.keras.layers import Conv2D, MaxPooling2D, Flatten, Dense
```

# Create a Sequential model
classifier = Sequential()

# Add the first convolutional layer
classifier.add(Conv2D(32, (3, 3), input_shape=(175, 130, 3), activation='relu'))
classifier.add(MaxPooling2D(pool_size=(2, 2)))

# Add the second convolutional layer
classifier.add(Conv2D(64, (3, 3), activation='relu'))
classifier.add(MaxPooling2D(pool_size=(2, 2)))

# Flatten the output from the convolutional layers
classifier.add(Flatten())

# Add a fully connected layer
classifier.add(Dense(128, activation='relu'))

# Add the output layer for binary classification (1 neuron with sigmoid activation)
classifier.add(Dense(1, activation='sigmoid'))

# Compile the model
classifier.compile(optimizer='adam', loss='binary_crossentropy', metrics=['accuracy'])
```

In this corrected code, we have a single instance of each convolutional layer (with the parameters you specified) followed by the appropriate pooling layers and other layers for your binary classification task.
